comment/models/comments.py

```diff
@@ -69,7 +69,8 @@ def _set_unique_urlhash(self):
 
     def _set_email(self):
         if self.user:
-            self.email = self.user.email
+            if getattr(self.user, 'email', False):
```

Comment:
You can probably replace this with
```suggestion
            if hasattr(self.user, 'email'):
```

Reply:
also, with this change you will have to change the field `email` on the model to accept `null` values as well. The existing tests might pass because, as of now, you have not written a test to verify this change.

Reply:
What if i add blank email? Blank is set on EmailField
```suggestion
            if getattr(self.user, 'email'):
                self.email = self.user.email
            else:
                self.email = ''
```

Reply:
Sorry, I didn't remember that I had set `null=True` on the `EmailField` for the model. Why would you want to set it again when we have already set it inside the model? I suspect `django` could already do this for you.

Also, in your suggestion I don't quite see the point of prefering `getattr` over `hasattr` where all you want to do is check whether the `user` object has the `email` attribute or not. Am I missing a point here?

Reply:
I am sorry getattr was a typo. I'll push the changes in an hour

Reply:
no issues. there is no need to be sorry, everyone should be allowed to make mistakes.